source/slang/slang.cpp

```diff
@@ -3481,16 +3457,39 @@ SLANG_NO_THROW SlangResult SLANG_MCALL ComponentType::getEntryPointCode(
     return artifact->loadBlob(ArtifactKeep::Yes, outCode);
 }
 
-SLANG_NO_THROW SlangResult SLANG_MCALL ComponentType::computeDependencyBasedHash(slang::Checksum* outHashCode)
+SLANG_NO_THROW void SLANG_MCALL ComponentType::computeDependencyBasedHash(
+    SlangInt entryPointIndex,
+    SlangInt targetIndex,
+    slang::Hash* outHash)
 {
-    SLANG_UNUSED(outHashCode);
-    return SLANG_E_NOT_AVAILABLE;
+    HashBuilder builder;
+    builder.hashGen.init(&builder.context);
+
+    // A note on enums that may be hashed in as part of the following two function calls:
+    // 
+    // While enums are not guaranteed to be encoded the same way across all versions of
+    // the compiler, part of hashing the linkage is hashing in the compiler version.
+    // Consequently, any encoding differences as a result of different compiler versions
+    // will already be reflected in the resulting hash.
+    getLinkage()->computeDependencyBasedHash(builder, targetIndex);
```

Comment:
should be `getLinkage()->updateDependencyBasedHash()`, this method in the linkage should just be internal. All internal methods are named `update*` according to the implicit convention here.